udemy_exercises/songs/Album.java

```diff
@@ -1,19 +1,52 @@
-package udemy_exercises.songs;
+import java.util.ArrayList;
+import java.util.LinkedList;
 
-class Album {
+public class Album {
+    private String name;
+    private String artist;
+    private ArrayList<Song> songs;
 
-}
+    public Album(String name, String artist) {
+        this.name = name;
+        this.artist = artist;
+        this.songs = new ArrayList<Song>();
+    }
+
+    public boolean addSong(String title, double duration) {
+        if (findSong(title) == null) {
+            this.songs.add(new Song(title, duration));
+            return true;
+        }
+        return false;
+    }
+
+    private Song findSong(String title) {
+        for (Song checkedSong : this.songs) {
+            if (checkedSong.getTitle().equals(title)) {
+                return checkedSong;
+            }
+        }
+        return null;
+    }
 
-/*
-**Song Player w/ Java Classes**
-     - **Album class** 
-       - Playlist of **Songs**.
-       - addPlayist() (songs must appear in the list in the order by which they were added)
-       - stop()
-       - nextSong()
-       - previousSong()
-       - repeatSong()
-       - showPlaylist()
-       - removePlaylist()
-     - **A song must exist in an album before it can be added to the playlist (so you can only play songs that you own.)**
-*/
\ No newline at end of file
+    public boolean addToPlayList(int trackNumber, LinkedList<Song> playList) {
+        int index = trackNumber - 1;
+        if ((index > 0) && (index <= this.songs.size())) {
```

Comment:
Since an index is 0 based, and the 'if' statement requires index to be greater than zero, how is the first track in the LinkedList ever accessed?  At the same time, what will happen if the trackNumber argument is set to '9' for LinkedList with '8' items?  In this case, index (which will be '8') will be equal to songs.size(), while the last item in the list is actually at index '7'.